src/createServer.js

```diff
@@ -1,3 +1,108 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const { errorMessage, caseName } = require('./errorMessage');
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    res.writeHead(200, { 'Content-type': 'application/json' });
+
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = normalizedUrl.pathname.slice(1);
+    const targetCase = normalizedUrl.searchParams.get('toCase');
+
+    if (!originalText && !targetCase) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.missingText },
+          { message: errorMessage.missingParam },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!originalText && !caseName.includes(targetCase)) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.caseNotSupported },
+          { message: errorMessage.missingText },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!targetCase) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          {
+            message: errorMessage.missingParam,
+          },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!originalText) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          {
+            message: errorMessage.missingText,
+          },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!caseName.includes(targetCase)) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.caseNotSupported },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    };
```

Comment:
Create a separate function that will take the `targetCase `and the `originalText` as arguments, validate them, and return the errors array.
After that check if you have any errors in the errors array and update the response
You have a lot of repeated logic.


`If something is not correct, you should respond with 400 status, Bad request statusText`

You've forgotten about the status text.